src/AppBundle/Tests/FormCrawler/ValidationTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+
+namespace AppBundle\Tests\Validation;
+
+use AppBundle\Tests\BaseWebTestCase;
+use PhpCollection\Set;
+use Symfony\Component\DomCrawler\Crawler;
+
+class ValidationTest extends BaseWebTestCase
+{
+    const urls = array(
+        "/assistenter",
+        "/resetpassord",
+        "/team/application/1",
+        "/kontrollpanel/deltakerhistorikk/rediger/1",
+        "/kontrollpanel/hovedstyret/rediger_medlem/2",
+        "/kontrollpanel/teamadmin/oppdater/teamhistorie/1",
+        "/kontrollpanel/semesteradmin/avdeling/opprett/1",
+        "/kontrollpanel/brukeradmin/opprett/1",
+        "/kontrollpanel/teamadmin/stillinger",
+        "/kontrollpanel/teamadmin/update/6",
+        "/kontrollpanel/hovedstyret/oppdater",
+        "/kontrollpanel/intervju/settopp/6",
+        "/kontrollpanel/intervju/conduct/6",
+        "/kontrollpanel/intervju/admin_assign_co_interviewer/3",
+        "/kontrollpanel/semesteradmin/update/2",
+        "/kontrollpanel/vikar/rediger/1",
+        "/kontrollpanel/intervju/skjema/2",
+        "/kontrollpanel/avdelingadmin/update/1",
+        "/kontrollpanel/linje/1",
+        "/kontrollpanel/skoleadmin/oppdater/1",
+        "/kontrollpanel/skoleadmin/tildel/skole/1",
+        "/kontrollpanel/nyhetsbrev/send/1",
+        "/nyhetsbrev/1",
+        "/kontrollpanel/nyhetsbrev/opprett",
+        "/kontrollpanel/undersokelse/opprett",
+        "/kontrollpanel/sponsors/edit",
+        "/kontrollpanel/utlegg/rediger/4",
+        "/kontrollpanel/attest/signatur",
+    );
+
+    const formElements = array(
+                'input',
+                'select',
+                'textarea',
+                'datalist'
+    );
+    const maxNumRedirects = 20;
+
+    public function testValidation()
```

Comment:
Liker tanken med å ha en crawler som "automatisk" leter etter bad behavior under form submission. For meg er det likevel litt uklart hva det er som testes her siden det ikke er noen asserts som beskriver hva som er forventet utfall. Hva er egentlig målet med denne testen?

Reply:
Den testen er litt dirty, det skal sies.

Den går gjennom linkene og prøver å putte inn `null` i forskjellige felter. Hvis vi mangler validering noen plasser det burde være det, eller noe annet i koden ikke tar hensyn til at input dataene kan være `null`, resulterer dette i en 500 feil, akkurat som det ville gjort i nettleseren. Testen vil da feile med error og avbrytes. Altså trengs det ingen asserts for å plukke det opp. Den printer også en stacktrace. 

Jeg brukte egentlig bare testen som en quick n dirty måte å få banka ut valideringsfeil fra formene på sida, uten å måtte teste alle manuelt. Den var veldig nyttig til akkurat det, men siden den er både treg og lite elegant vurderer jeg sterkt å bare fjerne den. Hva syns du?

Reply:
Ja, det var jo en lur måte å finne valideringsfeil i dette tilfellet, men jeg er litt redd for at den både kan gi litt ustabile resultater og bruke lang tid på å kjøre. Derfor er den kanskje ikke like praktisk å bruke senere som det var i dette tilfellet. Vi kan heller se på hvordan vi kan teste forms med unit tests :)

Reply:
Nei ikke sant. Da bare fjerner jeg den for nå. 

Legger en kopi [her](https://gist.github.com/sigtot/51de9a8dededb61a3bc55c05e40efa35) hvis noen andre vil se på den.